internal/operators/odf/common.go

```diff
@@ -0,0 +1,42 @@
+package odf
+
+import (
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+func getODFDeploymentMode(cluster *models.Cluster, odfMinimumHosts int64) odfDeploymentMode {
+	masterHosts, workerHosts, autoAssignHosts := common.GetHostsByEachRole(cluster, true)
+
+	masterCount := len(masterHosts)
+	workerCount := len(workerHosts)
+	autoAssignCount := len(autoAssignHosts)
+	hostCount := masterCount + workerCount + autoAssignCount
+
+	// To keep compatability with the behaviour until now.
+	if hostCount == common.MinMasterHostsNeededForInstallationInHaMode && workerCount == 0 {
+		return compactMode
+	}
+
+	if masterCount == hostCount && masterCount >= int(odfMinimumHosts) {
+		return compactMode
+	}
+
+	if masterCount >= common.MinMasterHostsNeededForInstallationInHaMode && workerCount >= int(odfMinimumHosts) {
+		return standardMode
+	}
+
+	// To be determined, the cluster is not yet in a valid form.
+	return unknown
+}
+
+// To keep compatability with the behaviour until now.
+// TODO - remove this once two control plane nodes OpenShift is implemented
+func isAutoAssignmentAllowed(cluster *models.Cluster) bool {
```

Comment:
Shouldn't this part of the following function (`willHostPotentiallyRunODF`)?

Reply:
It made more sense to me to split it to a different function, as it is more of a validation utility rather then a function who checks a condition

Reply:
But the code usage is always one after the other used in the same way.
If we group these functions (can be a third one just grouping them) it will make the code neater and DRY-er. Don't you think?